rotkehlchen/chain/ethereum/modules/superchain_bridge/decoder.py

```diff
@@ -115,16 +130,37 @@ def _decode_prove_withdrawal(self, context: DecoderContext) -> DecodingOutput:
             balance=Balance(),
             location_label=context.transaction.from_address,
             notes=f'Prove optimism bridge withdrawal 0x{withdrawal_hash}',
-            counterparty=CPT_OPTIMISM,
+            counterparty=self.get_counterparty_details(target_chain).identifier,
             address=context.tx_log.address,
         )
         return DecodingOutput(event=event)
 
+    @staticmethod
+    def get_target_chain(address: ChecksumEvmAddress) -> ChainID:
+        """Get the target chain for a bridge address."""
+
+        for addresses, chain_id in BRIDGE_ADDRESSES_TO_TARGET_CHAINS.items():
+            if address in addresses:
+                return chain_id
+
+        raise ValueError(f'Unsupported bridge address: {address}')
+
+    @staticmethod
+    def get_counterparty_details(chain: ChainID) -> CounterpartyDetails:
```

Comment:
Don't do it like this. There is no reason to create a condtional checker that performs an if at every time you need this. Instead initialize the bridge's counterparty at the constructor.